### PR TITLE
Fix additive handling for Tribe Commerce coming from ticket block

### PR DESCRIPTION
**Pull request by sc0ttkclark**
https://central.tri.be/issues/137988

### CODE REVIEW

File: src/Tribe/Commerce/PayPal/Main.php [hunk 1]
```diff
@@ -251,7 +251,7 @@ public function bind_implementations() {
 		tribe_singleton( 'tickets.commerce.paypal.view', 'Tribe__Tickets__Commerce__PayPal__Tickets_View' );
 		tribe_singleton( 'tickets.commerce.paypal.handler.ipn', 'Tribe__Tickets__Commerce__PayPal__Handler__IPN', array( 'hook' ) );
 		tribe_singleton( 'tickets.commerce.paypal.handler.pdt', 'Tribe__Tickets__Commerce__PayPal__Handler__PDT', array( 'hook' ) );
-		tribe_singleton( 'tickets.commerce.paypal.gateway', 'Tribe__Tickets__Commerce__PayPal__Gateway', array( 'hook', 'build_handler' ) );
+		tribe_singleton( 'tickets.commerce.paypal.gateway', 'Tribe__Tickets__Commerce__PayPal__Gateway', array( 'build_handler' ) );
```
Comment (sc0ttkclark) [inline src/Tribe/Commerce/PayPal/Main.php]: Removing the old PayPal add to cart functionality that was often conflicting with the handling we built in Commerce/Cart.php

File: src/views/registration-js/content.php [hunk 2]
```diff
@@ -101,6 +101,7 @@
 			<input type="hidden" name="tribe_tickets_ar" value="1" />
 			<input type="hidden" name="tribe_tickets_ar_page" value="1" />
 			<input type="hidden" name="tribe_tickets_ar_data" value="" id="tribe_tickets_ar_data"  />
+			<input type="hidden" name="tribe_tickets_provider" value="<?php echo esc_attr( $provider ); ?>"  />
```
Comment (sc0ttkclark) [inline src/views/registration-js/content.php]: Makes sure we always send provider

File: src/Tribe/Commerce/Cart.php [hunk 3]
```diff
@@ -34,15 +34,67 @@ public function process_cart() {
 				$data = stripslashes( $data );
 				$data = json_decode( $data, true );
 			}
+
+			$data = array_merge( $_POST, $data );
 		}
 
-		$post_id  = isset( $data['tribe_tickets_post_id'] ) ? $data['tribe_tickets_post_id'] : null;
-		$provider = isset( $data['tribe_tickets_provider'] ) ? $data['tribe_tickets_provider'] : null;
+		$post_id  = isset( $data['tribe_tickets_post_id'] ) ? absint( $data['tribe_tickets_post_id'] ) : null;
```
Comment (sc0ttkclark) [inline src/Tribe/Commerce/Cart.php]: Fix sanitization

File: src/Tribe/Commerce/Cart.php [hunk 4]
```diff
@@ -34,15 +34,67 @@ public function process_cart() {
 				$data = stripslashes( $data );
 				$data = json_decode( $data, true );
 			}
+
+			$data = array_merge( $_POST, $data );
```
Comment (sc0ttkclark) [inline src/Tribe/Commerce/Cart.php]: Make sure we still get other vars coming through

File: src/Tribe/Commerce/Cart.php [hunk 5]
```diff
@@ -34,15 +34,67 @@ public function process_cart() {
...
 		}
 
-		$post_id  = isset( $data['tribe_tickets_post_id'] ) ? $data['tribe_tickets_post_id'] : null;
-		$provider = isset( $data['tribe_tickets_provider'] ) ? $data['tribe_tickets_provider'] : null;
+		$post_id  = isset( $data['tribe_tickets_post_id'] ) ? absint( $data['tribe_tickets_post_id'] ) : null;
+		$provider = isset( $data['tribe_tickets_provider'] ) ? sanitize_text_field( $data['tribe_tickets_provider'] ) : null;
 		$tickets  = isset( $data['tribe_tickets_tickets'] ) ? $data['tribe_tickets_tickets'] : null;
 		$meta     = isset( $data['tribe_tickets_meta'] ) ? $data['tribe_tickets_meta'] : null;
 
+		if ( null === $provider ) {
```
Comment (sc0ttkclark) [inline src/Tribe/Commerce/Cart.php]: Provider detection if not sent by expected parameter names

File: src/Tribe/Commerce/Cart.php [hunk 6]
```diff
@@ -34,15 +34,67 @@ public function process_cart() {
...
+			'tribe-commerce',
+			'tribe_tpp_attendees',
+			'Tribe__Tickets__Commerce__PayPal__Main',
+		];
+
+		$is_tribe_commerce = in_array( $provider, $tribe_commerce_providers, true );
+
+		// Simplify provider for Tribe Commerce.
+		if ( $is_tribe_commerce ) {
+			$provider = 'tribe-commerce';
```
Comment (sc0ttkclark) [inline src/Tribe/Commerce/Cart.php]: Enforces provider name for tribe-commerce for better uniformity further into Cart handling/hooks

File: src/Tribe/Commerce/Cart.php [hunk 7]
```diff
@@ -34,15 +34,67 @@ public function process_cart() {
...
+		}
+
 		// On AR Page, we use replace logic, not additive.
 		$is_ar_modal = empty( $_POST['tribe_tickets_ar_page'] );
+		$additive    = $is_ar_modal && ! $is_tribe_commerce;
+
+		if ( $is_tribe_commerce && null === $post_id ) {
```
Comment (sc0ttkclark) [inline src/Tribe/Commerce/Cart.php]:
```suggestion
		if ( $is_tribe_commerce ) {
```

File: src/Tribe/Commerce/Cart.php [hunk 8]
```diff
@@ -34,15 +34,67 @@ public function process_cart() {
...
+		if ( $is_tribe_commerce ) {
+			$provider = 'tribe-commerce';
+		}
+
 		// On AR Page, we use replace logic, not additive.
 		$is_ar_modal = empty( $_POST['tribe_tickets_ar_page'] );
+		$additive    = $is_ar_modal && ! $is_tribe_commerce;
+
+		if ( $is_tribe_commerce ) {
+			if ( null === $post_id ) {
```
Comment (sc0ttkclark) [inline src/Tribe/Commerce/Cart.php]: Detect post ID from the request if it's not submitted with POST

File: src/Tribe/Commerce/Cart.php [hunk 9]
```diff
@@ -34,15 +34,67 @@ public function process_cart() {
...
+				if ( ! empty( $_GET['tribe_tickets_post_id'] ) ) {
+					// Get post ID from current URL parameter.
+					$post_id = absint( $_GET['tribe_tickets_post_id'] );
+				} else {
+					// Detect post ID for Tribe Commerce from the first ticket (no cart, we only support one post at a time).
```
Comment (sc0ttkclark) [inline src/Tribe/Commerce/Cart.php]: Detect post ID from the first ticket in the cart if it's not submitted with POST

File: src/Tribe/Commerce/Cart.php [hunk 10]
```diff
@@ -34,15 +34,67 @@ public function process_cart() {
...
+
+						if ( $ticket ) {
+							$post_id = $ticket->get_event_id();
+						}
+					}
+				}
+			}
+
+			if ( null === $meta && $is_ar_modal ) {
+				$meta = [];
```
Comment (sc0ttkclark) [inline src/Tribe/Commerce/Cart.php]: If not coming from AR page, enforce meta reset for Tribe Commerce 'temporary cart'

File: src/Tribe/Commerce/Cart.php [hunk 11]
```diff
@@ -34,15 +34,67 @@ public function process_cart() {
...
 
-		$post_id  = isset( $data['tribe_tickets_post_id'] ) ? $data['tribe_tickets_post_id'] : null;
-		$provider = isset( $data['tribe_tickets_provider'] ) ? $data['tribe_tickets_provider'] : null;
+		$post_id  = isset( $data['tribe_tickets_post_id'] ) ? absint( $data['tribe_tickets_post_id'] ) : null;
+		$provider = isset( $data['tribe_tickets_provider'] ) ? sanitize_text_field( $data['tribe_tickets_provider'] ) : null;
 		$tickets  = isset( $data['tribe_tickets_tickets'] ) ? $data['tribe_tickets_tickets'] : null;
 		$meta     = isset( $data['tribe_tickets_meta'] ) ? $data['tribe_tickets_meta'] : null;
 
+		if ( null === $provider ) {
+			if ( ! empty( $_GET['provider'] ) ) {
```
Comment (Camwyn) [inline src/Tribe/Commerce/Cart.php]: can we use `tribe_get_request_var()` here instead of the if/else?

**Pull request merged**